### ISSUE TITLE
Update contributor guide and PR template

### ISSUE THREAD

**Issue by chrisiacovella** [choderalab/modelforge#260]
Currently, our contributor guide is rather minimal. 

I propose we update this to be a bit more specific, in particular noting users should first create an issue before moving forward with changes.   We already discuss some of the formatting aspects, but we can expand upon that a bit, e.g., here is my proposed list of guidelines (which I think are also useful to remind US what we should do each time, particularly the part about updating/creating appropriate .rst files and the wiki pages).

- Open an issue on github to discuss the proposed changes before starting work.
- Fork the project and create a new branch for your changes.
- Use the `PEP8 <https://pep8.org/>`_ style guide for Python code.
- Use `black <https://black.readthedocs.io/en/stable/>`_ for code formatting.
- Use Python type hints to improve code readability and maintainability.
- Include docstrings formatted using the `numpydoc <https://numpydoc.readthedocs.io/en/latest/>`_ style.
- Include unit tests for new code, including both positive and negative tests when applicable, and test edge cases.
- Include/update documentation in the form of .rst files (in the docs folder).
- Ensure that the code is well-commented.
- If applicable, include examples of how to use the new code.
- For more complex additions (e.g., new NNPs) create a new wiki page to explain the underlying algorithms. 

Related, I think we should put some of these requirements into the PR template, so we remind ourselves to do them.

New PR template

```
## Pull Request Summary
Provide a brief description of the PR's purpose here.

### Key changes
Notable points that this PR has either accomplished or will accomplish.
 - [ ] Change 1

### Questions
 - [ ] Question 1

### Associated Issue(s)
 - [ ] Issue 1

## Pull Request Checklist
 - [ ] Issue(s) raised/addressed and linked
 - [ ] Includes appropriate unit test(s)
 - [ ] Appropriate docstring(s) added/updated
 - [ ] Appropriate .rst doc file(s) added/updated
 - [ ] PR is ready for review
```

In chatting with @ianmkenney, we may want to also include a rule to enforce that code coverage doesn't drop as a result of a PR.  I always try to look at codecov before merging to see if my new code is actually getting tested, but it might be good to have something that automatically alerts us if we have failed at such things
